lib/ctbl.gd

```diff
@@ -3884,24 +3886,65 @@ DeclareAttributeSuppCT( "SourceOfIsoclinicTable", IsNearlyCharacterTable,
 ##
 #F  CharacterTableOfNormalSubgroup( <ordtbl>, <classes> )
 ##
+##  <#GAPDoc Label="CharacterTableOfNormalSubgroup">
 ##  <ManSection>
 ##  <Func Name="CharacterTableOfNormalSubgroup" Arg='ordtbl, classes'/>
 ##
 ##  <Description>
-##  returns the restriction of the ordinary character table <A>ordtbl</A>
-##  to the classes in the list <A>classes</A>.
-##  <P/>
-##  In most cases, this table is only an approximation of the character table
-##  of this normal subgroup, and some classes of the normal subgroup must be
-##  split (see&nbsp;<Ref Func="CharacterTableSplitClasses"/>) in order to get
-##  a character table.
-##  The result is only a table in progress then
-##  (see&nbsp;<Ref Sect="Character Table Categories"/>).
-##  <P/>
-##  If the classes in <A>classes</A> need not to be split then the result is
-##  a proper character table.
+##  Let <A>ordtbl</A> be the ordinary character table of a group <M>G</M>,
+##  say, and <A>classes</A> be a list of class positions for this table.
+##  If the classes given by <A>classes</A> form a normal subgroup <M>N</M>,
+##  say, of <M>G</M> and if these classes are conjugacy classes of <M>N</M>
+##  then this function returns the character table of <M>N</M>.
+##  In all other cases, the function returns <K>fail</K>.
+##  <P/>
+##  <Example><![CDATA[
+##  gap> t:= CharacterTable( "Symmetric", 4 );
+##  CharacterTable( "Sym(4)" )
+##  gap> nsg:= ClassPositionsOfNormalSubgroups( t );
+##  [ [ 1 ], [ 1, 3 ], [ 1, 3, 4 ], [ 1 .. 5 ] ]
+##  gap> rest:= List( nsg, c -> CharacterTableOfNormalSubgroup( t, c ) );
+##  [ CharacterTable( "Rest(Sym(4),[ 1 ])" ), fail, fail, 
+##    CharacterTable( "Rest(Sym(4),[ 1 .. 5 ])" ) ]
+##  ]]></Example>
+##  <P/>
+##  Here is a nontrivial example.
+##  We use <Ref Func="CharacterTableOfNormalSubgroup"/> for computing the
+##  two isoclinic variants of <M>2.A_5.2</M>.
+##  <P/>
+##  <Example><![CDATA[
+##  gap> g:= SchurCoverOfSymmetricGroup( 5, 3, 1 );;
+##  gap> c:= CyclicGroup( 4 );;
+##  gap> dp:= DirectProduct( g, c );;
+##  gap> diag:= First( Elements( Centre( dp ) ), 
```

Comment:
Optional: Using `Elements` here is not needed, it's perfectly fine to do `First( Centre( dp ), ...)`